lib/OpenQA/Git.pm

```diff
@@ -171,4 +176,36 @@ sub is_workdir_clean ($self) {
     return $r->{status};
 }
 
+sub cache_ref ($self, $ref, $url, $relative_path, $output_file, $allow_arbitrary_url_fetch = 1) {
+    $self->app->log->debug(
+        "Checking out $relative_path from url '" . ($url // '') . "' rev '" . ($ref // '') . "' to $output_file");
+    # checkout git versioned file <$relative_path> of from repo at version <$ref> and write it to <$output_file>
+    # returns undef on success - else the error message
+    if (-f $output_file) {
+        eval { path($output_file)->touch };
+        return $@ ? $@ : undef;
+    }
+
+    # Now check if the ref is already present in the repo
+    my @git = $self->_prepare_git_command(1);
+    my $res = run_cmd_with_log_return_error [@git, 'cat-file', '-t', $ref];
```

Comment:
```suggestion
    my $res = $self->_run_cmd(['cat-file', '-t', $ref], {expected_return_codes => {0 => 1, 1 => 1, 128 => 1}});
```
This is now longer, and I think it can be made shorter by making `expected_return_codes` an array. Not sure why it was made as a hash in the fist place.

Reply:
I updated it to:
```perl
my $res = $self->_run_cmd(['cat-file', '-t', $ref], \%CHECK_OPTIONS);